### PR TITLE
TP2000-928 create origins from edit quota

### PR DESCRIPTION
**Pull request by a-gleeson**
# TP2000-928 create origins from edit quota

## Why
Following a change to Steel Safeguards geographical areas to reflect the “pots” that can be drawn on, the ability to add additional geographical origins to a quota order number is needed to manage origins correctly in the user interface.

## What
- added a create new origin button to the edit quota form
- added origin page added ( similar to edit origin page)
- added origin confirm create page added ( similar to edit origin page)
- added return to quota edit page button to create and edit confirmation page
- tests to check invalid origins can't be created

image
image
image
image

### CODE REVIEW

File: quotas/jinja2/quota-origins/confirm-create.jinja [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{% extends "common/confirm_create.jinja" %}
+{% from "components/breadcrumbs.jinja" import breadcrumbs %}
+{% from "components/button/macro.njk" import govukButton %}
+
+{% set page_title = "Create quota order number origin details" %}
```
Comment (eadpearce) [inline quotas/jinja2/quota-origins/confirm-create.jinja:5]: This reads a bit clunky. Maybe needs some finessing by a content person?

File: quotas/tests/test_views.py [hunk 2]
```diff
@@ -772,3 +772,118 @@ def test_quota_edit_origin_exclusions_remove(
...
+
+
+def test_quota_create_origin_outwith_quota_period(
+    valid_user_client,
+    approved_transaction,
+    geo_group1,
+    date_ranges,
+):
+    """Checks that for a quota that you cannot create a quota origin that lies
+    outside the."""
```
Comment (eadpearce) [inline quotas/tests/test_views.py:819]: Incomplete docstring?

File: quotas/jinja2/quota-origins/confirm-create.jinja (hunk 1 above)
Comment (a-gleeson) [reply quotas/jinja2/quota-origins/confirm-create.jinja:5]: You're right, I'll check that

File: quotas/jinja2/quota-origins/create.jinja [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+{% set page_title = "Create a new quota origin" %}
+
+{% block content %}
+  <h1 class="govuk-heading-xl">{{ page_title }}</h1>
+  {% if page_label %}
+    <div class="govuk-hint"> {{ page_label }}</div>
+  {% endif %}
+  <p class="govuk-body">
+    Find out more about <a class="govuk-link" href="https://data-services-help.trade.gov.uk/tariff-application-platform/tariff-policy/origin-quotas/">quota origins</a>.
```
[the remark this answers is not in the document]
Comment (a-gleeson) [reply quotas/jinja2/quota-origins/create.jinja:11]: That makes sense to me, I'll check how I can refactor this content

File: quotas/tests/test_views.py (hunk 2 above)
Comment (a-gleeson) [reply quotas/tests/test_views.py:819]: Good catch!

File: quotas/jinja2/quota-origins/create.jinja (hunk 3 above)
[the remark this answers is not in the document]
Comment (a-gleeson) [reply quotas/jinja2/quota-origins/create.jinja:11]: I spotted that this should be included in the page_label so will try refactor so this content is set in the view

File: quotas/jinja2/quota-origins/confirm-create.jinja (hunk 1 above)
Comment (a-gleeson) [reply quotas/jinja2/quota-origins/confirm-create.jinja:5]: removed to use standard confirm-create page title to make consitent with others

File: quotas/jinja2/quota-origins/create.jinja (hunk 3 above)
[the remark this answers is not in the document]
Comment (a-gleeson) [reply quotas/jinja2/quota-origins/create.jinja]: Also updated geo area create as it was similar

**Review by eadpearce — APPROVED** [review]

**Pull request merged**